autogen/oai/mistral.py

```diff
@@ -86,34 +99,48 @@ def parse_params(self, params: Dict[str, Any]) -> Dict[str, Any]:
         )
         mistral_params["random_seed"] = validate_parameter(params, "random_seed", int, True, None, False, None)
 
+        # TODO
+        if params.get("stream", False):
+            warnings.warn(
+                "Streaming is not currently supported, streaming will be disabled.",
+                UserWarning,
+            )
+
         # 3. Convert messages to Mistral format
         mistral_messages = []
         tool_call_ids = {}  # tool call ids to function name mapping
         for message in params["messages"]:
             if message["role"] == "assistant" and "tool_calls" in message and message["tool_calls"] is not None:
                 # Convert OAI ToolCall to Mistral ToolCall
-                openai_toolcalls = message["tool_calls"]
-                mistral_toolcalls = []
-                for toolcall in openai_toolcalls:
-                    mistral_toolcall = ToolCall(id=toolcall["id"], function=toolcall["function"])
-                    mistral_toolcalls.append(mistral_toolcall)
-                mistral_messages.append(
-                    ChatMessage(role=message["role"], content=message["content"], tool_calls=mistral_toolcalls)
-                )
+                mistral_messages_tools = []
+                for toolcall in message["tool_calls"]:
+                    mistral_messages_tools.append(
+                        ToolCall(
+                            id=toolcall["id"],
+                            function=FunctionCall(
+                                name=toolcall["function"]["name"],
+                                arguments=json.loads(toolcall["function"]["arguments"]),
+                            ),
+                        )
+                    )
+
+                mistral_messages.append(AssistantMessage(content="", tool_calls=mistral_messages_tools))
```

Comment:
Can mistral accept content when type is a tool_call?

Reply:
Hey @yiranwu0, thanks so much for reviewing this. Yes, this works fine.